src/components/menu-top/LoadShow.vue

```
@@ -0,0 +1,130 @@
+          v-if="file"
+          class="file-name"
+        >
+          {{ file.name }}
+        </span>
+      </b-field>
+      <span
+        v-if="parseError"
+      >
+        <b-message
+          title="Error"
+          type="is-danger"
+          aria-close-label="Close message"
+        >
+          {{ parseError }}
+        </b-message>
+      </span>
+      <span
+        v-if="show"
+      >
+        <b-field>
+          Marchers: {{ numMarchers }}
+        </b-field>
+        <b-field>
+          Sheets: {{ numSheets }}
+        </b-field>
+      </span>
+    </section>
+
+    <footer class="modal-card-foot">
+      <b-button
+        type="is-success"
+        data-test="file-modal--import"
+        :disabled="!showSet"
+        @click="setShow();$parent.close()"
+      >
+        Import
+      </b-button>
+      <b-button
+        type="is-primary"
+        data-test="file-modal--close"
+        @click="$parent.close()"
+      >
+        Close
+      </b-button>
+    </footer>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { LoadShow } from '@/models/util/LoadShow';
+import Show from '@/models/Show';
+
+export default Vue.extend({
+  name: 'LoadShow',
+  data: (): {
+    file: Blob | null;
+    show: Show | null;
+    showSet: boolean;
+    parseError: string;
+  } => ({ file: null, show: null, showSet: false, parseError: '' }),
+  computed: {
+    numMarchers(): number {
+      return this.show ? this.show.dotLabels.length : 0;
+    },
+    numSheets(): number {
+      return this.show ? this.show.stuntSheets.length : 0;
+    },
+  },
+  methods: {
+    loadShow: function() {
+      if (this.file === null) {
+        return;
+      }
+      const reader = new FileReader();
+      reader.onload = () => {
+        // The file's text will be printed here
+        if (reader.result && reader.result instanceof ArrayBuffer) {
+          try {
+            const show = LoadShow(reader.result);
+            if (show) {
```

Note that this will silently fail if there is no `show`. Similar comment if there is no `reader.result`

---

Now that we aren't able to return null, we should be able to get rid of the check and just put things in place; an exception should be thrown in failure cases.

On the case when there's no `reader.result`, I'll manufacture an error.

---

Done!